packages/smooth_app/lib/cards/product_cards/knowledge_panels/knowledge_panel_table_card.dart

```diff
@@ -78,32 +121,180 @@ class KnowledgePanelTableCard extends StatelessWidget {
+          }
+        }
+        index++;
+      }
+    }
+    // We now allocate width to each column as follows:
+    // [availableWidth] / [column's largest cell width] * [totalMaxColumnWidth].
+    final int totalMaxColumnWidth =
+        columnMaxLength.reduce((int sum, int width) => sum + width);
+
+    final List<List<Widget>> rowsWidgets = <List<Widget>>[];
+    for (final List<TableCell> row in rows) {
+      final List<Widget> rowWidgets = <Widget>[];
+      int index = 0;
+      for (final TableCell cell in row) {
+        final double cellWidth =
+            availableWidth / totalMaxColumnWidth * columnMaxLength[index++];
+        rowWidgets.add(
+          _buildTableCellWidget(
+            context: context,
+            cell: cell,
+            cellWidth: cellWidth,
+          ),
+        );
+      }
+      rowsWidgets.add(rowWidgets);
+    }
+    return rowsWidgets;
+  }
+
+  Widget _buildTableCellWidget({
     required BuildContext context,
-    required String text,
+    required TableCell cell,
     required double cellWidth,
-    Color? textColor,
-    bool isHeader = false,
   }) {
     EdgeInsetsGeometry padding =
         const EdgeInsets.only(bottom: VERY_SMALL_SPACE);
     // header cells get a bigger vertical padding.
-    if (isHeader) {
+    if (cell.isHeader) {
       padding = const EdgeInsets.symmetric(vertical: SMALL_SPACE);
     }
     TextStyle style = Theme.of(context).textTheme.bodyText2!;
-    if (textColor != null) {
-      style = style.apply(color: textColor);
+    if (cell.color != null) {
+      style = style.apply(color: cell.color);
     }
-    return Padding(
-      padding: padding,
-      child: SizedBox(
+    Widget textWidget;
+    if (!cell.isHeader || cell.columnGroup!.columns.length == 1) {
+      // non-header cells and columnGroups with a single column are simple html text widgets.
+      textWidget = SizedBox(
         width: cellWidth,
         child: HtmlWidget(
-          text,
+          cell.text,
           textStyle: style,
         ),
-      ),
+      );
+    } else {
+      // Now we finally render [ColumnGroup]s as drop down menus.
+      textWidget = SizedBox(
+        width: cellWidth,
+        child: DropdownButtonHideUnderline(
+          child: ButtonTheme(
+            child: DropdownButton<KnowledgePanelTableColumn>(
+              value: cell.columnGroup!.currentColumn,
+              items: cell.columnGroup!.columns
+                  .map((KnowledgePanelTableColumn column) {
+                return DropdownMenuItem<KnowledgePanelTableColumn>(
+                  value: column,
+                  child: Container(
+                    // 24 px buffer is to allow the dropdown arrow icon.
+                    constraints:
+                        BoxConstraints(maxWidth: cellWidth - 24).normalize(),
+                    child: Text(column.textForSmallScreens ?? column.text),
+                  ),
```

Comment:
I don't think `24px` is enough for a touchable area (and I'm not even sure it's `px` or `dp`):
* In the [Human Interface Guidelines](https://developer.apple.com/ios/human-interface-guidelines/visual-design/adaptivity-and-layout/), Apple recommends a minimum target size of 44 pixels (px) wide 44 pixels tall.
* In the [Android Material Design Guidelines](https://material.io/guidelines/components/buttons.html#buttons-style) it’s suggested that touch targets should be at least 48 x 48 dp*

cf. https://medium.com/@zacdicko/size-matters-accessibility-and-touch-targets-56e942adc0cc

Reply:
so the whole  container is a touchable element. 24 (dp?) is only needed to show the dropdown icon (which is 23.6 dp large).